src/kzg10/data_structures.rs

```diff
@@ -557,3 +562,35 @@ impl<E: PairingEngine> ToBytes for Proof<E> {
             .write(&mut writer)
     }
 }
+
+/// Opening proofs of a commitment on a large domain
+pub struct AmortizedProof<E: PairingEngine> {
+    /// This is a vector of commitments to the witness polynomials
+    /// over a domain 1, omega, omega^2, ..., omega^{n-1}
+    /// where omega is a primitive n'th root of unity
+    pub w: Vec<E::G1Projective>,
+    /// Scale factor whose multiplication is deferred until the proofs are combined
+    pub scale: E::Fr,
+}
+
+impl<E: PairingEngine> AmortizedProof<E> {
+    /// Combine opening proofs onto a subset of the domain
+    pub fn combine_at_evals(
+        &self,
+        range: &std::ops::Range<usize>, // Domain is omega^{start}, ..., omega^{end-1}
```

Comment:
maybe just replacing this with `start: usize, end: usize` would work better? Or do we need it to be a `Range`?

Reply:
Sure, I think that makes sense. Ideally it seems more rustic to use something like `Range` but it has [it's own issues](https://ridiculousfish.com/blog/posts/least-favorite-rust-type.html).